python/cuml/dask/ensemble/randomforestregressor.py

```diff
@@ -0,0 +1,327 @@
+#
+# Copyright (c) 2019, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from dask.distributed import Client, wait, default_client, get_worker
+from dask import delayed
+from cuml.common.handle import Handle
+import dask.dataframe as dd
+import numba.cuda
+import random
+from tornado import gen
+from toolz import first
+from cuml.ensemble import RandomForestRegressor as cuRFR
+import math
+    
+
+def parse_host_port(address):
+    if '://' in address:
+        address = address.rsplit('://', 1)[1]
+    host, port = address.split(':')
+    port = int(port)
+    return host, port
+
+@gen.coroutine
+def _extract_ddf_partitions(ddf):
+    """
+    Given a Dask cuDF, return a tuple with (worker, future) for each partition
+    """
+    client = default_client()
+    
+    delayed_ddf = ddf.to_delayed()
+    parts = client.compute(delayed_ddf)
+    yield wait(parts)
+    
+    key_to_part_dict = dict([(str(part.key), part) for part in parts])
+    who_has = yield client.who_has(parts)
+
+    worker_map = []
+    for key, workers in who_has.items():
+        worker = parse_host_port(first(workers))
+        worker_map.append((worker, key_to_part_dict[key]))
+
+    gpu_data = {worker:part for worker, part in worker_map}
+
+    yield wait(gpu_data)
+
+    raise gen.Return(gpu_data)
+
+
+class RandomForestRegressor:
+    """
+    Implements a multi-GPU Random Forest regressor model which fits multiple decision
+    tree classifiers in an ensemble.
+
+    Please check the single-GPU implementation of Random Forest regressor for more information
+    about the algorithm.
+
+     Parameters
+    -----------
+    n_estimators : int (default = 10)
+                   number of trees in the forest.
+    handle : cuml.Handle
+             If it is None, a new one is created just for this class.
+    split_algo : int (default = 1)
+                 0 for HIST, 1 for GLOBAL_QUANTILE and 2 for SPLIT_ALGO_END
+                 The type of algorithm to be used to create the trees.
+    split_criterion: int (default = 2)
```

Comment:
Same comment as classifier, strings would be better here, and also same question regarding naming of `split_criterion` vs `criterion`

Reply:
Following the SG RF.